src/vcBPA.cpp

```diff
@@ -924,86 +954,97 @@ vdkError vcBPA_ConvertReadPoints(vdkConvertCustomItem *pConvertInput, vdkPointBu
-    ptrdiff_t pointAttrOffset = ptrdiff_t(pBuffer->pointCount) * pBuffer->attributeStride;
-    for (uint32_t j = 0; j < pData->activeItem.pGrid->pBuffer->attributes.count; ++j)
-    {
-      vdkAttributeDescriptor &oldAttrDesc = pData->activeItem.pGrid->pBuffer->attributes.pDescriptors[j];
-      uint32_t attributeSize = (oldAttrDesc.typeInfo & (vdkAttributeTypeInfo_SizeMask << vdkAttributeTypeInfo_SizeShift));
+        // Get attribute old offset and pointer
+        uint32_t attrOldOffset = 0;
+        if (vdkAttributeSet_GetOffsetOfNamedAttribute(&pData->activeItem.pGrid->pBuffer->attributes, oldAttrDesc.name, &attrOldOffset) != vE_Success)
+          continue;
 
-      // Get attribute old offset and pointer
-      uint32_t attrOldOffset = 0;
-      if (vdkAttributeSet_GetOffsetOfNamedAttribute(&pData->activeItem.pGrid->pBuffer->attributes, oldAttrDesc.name, &attrOldOffset) != vE_Success)
-        continue;
+        void *pOldAttr = udAddBytes(pData->activeItem.pGrid->pBuffer->pAttributes, i * pData->activeItem.pGrid->pBuffer->attributeStride + attrOldOffset);
 
-      void *pOldAttr = udAddBytes(pData->activeItem.pGrid->pBuffer->pAttributes, i * pData->activeItem.pGrid->pBuffer->attributeStride + attrOldOffset);
+        // Get attribute new offset and pointer
+        uint32_t attrNewOffset = 0;
+        if (vdkAttributeSet_GetOffsetOfNamedAttribute(&pBuffer->attributes, oldAttrDesc.name, &attrNewOffset) != vE_Success)
+          continue;
 
-      // Get attribute new offset and pointer
-      uint32_t attrNewOffset = 0;
-      if (vdkAttributeSet_GetOffsetOfNamedAttribute(&pBuffer->attributes, oldAttrDesc.name, &attrNewOffset) != vE_Success)
-        continue;
+        void *pNewAttr = udAddBytes(pBuffer->pAttributes, pointAttrOffset + attrNewOffset);
 
-      void *pNewAttr = udAddBytes(pBuffer->pAttributes, pointAttrOffset + attrNewOffset);
+        // Copy attribute data
+        memcpy(pNewAttr, pOldAttr, attributeSize);
+      }
 
-      // Copy attribute data
-      memcpy(pNewAttr, pOldAttr, attributeSize);
+      // Displacement
+      float *pDisplacement = (float *)udAddBytes(pBuffer->pAttributes, pointAttrOffset + pData->displacementOffset);
+      *pDisplacement = (float)distance;
     }
 
-    // Displacement
-    float *pDisplacement = (float*)udAddBytes(pBuffer->pAttributes, pointAttrOffset + displacementOffset);
-    *pDisplacement = (float)distance;
-
-    ++pBuffer->pointCount;
-  }
+    if (pBuffer->pointCount == pBuffer->pointsAllocated)
+    {
+      pData->activeItem.pointIndex = maxPointIndex;
+      return vE_Success;
+    }
 
-  if (getNextGrid)
-  {
     pData->activeItem.pGrid->Deinit();
     pData->activeItem.oldGrid.Deinit();
     pData->activeItem.pGrid = nullptr;
     pData->activeItem.pointIndex = 0;
 
+    int i = 0;
     do
     {
       if (udSafeDeque_PopFront(pData->pConvertItemData, &pData->activeItem) == udR_Success)
         break;
+
+      if (++i > 25)
+        return vE_Success;
```

Comment:
This seems unnecessary to me, also, it really should `goto epilogue;` instead. Our coding standard is very clear on `return` statements, there's very few cases where we make exceptions on this.